app/models/host_status/configuration_status.rb

```diff
@@ -22,15 +22,17 @@ def out_of_sync?
     end
 
     def expected_report_interval
-      (reported_origin_interval + default_report_interval).minutes
+      (reported_origin_interval || default_report_interval).minutes
     end
 
     def reported_origin_interval
-      if last_report.origin &&
-         (interval = Setting[:"#{last_report.origin.downcase}_interval"])
+      if last_report.origin
+        if host.params.has_key? "#{last_report.origin.downcase}_interval"
+          interval = host.params["#{last_report.origin.downcase}_interval"]
+        else
+          interval = Setting[:"#{last_report.origin.downcase}_interval"]
+        end
         interval.to_i
```

Comment:
do we want to have `to_i` here ? that will mean if the origin interval is nil, we will have 0.
Don't we want to keep the fallback here and just remove the `+ ...` part in `expected_report_interval`?

Reply:
So you mean if we have e.g. `Salt` as `last.report.origin`, there is no `salt_interval` and we would have 0 as report interval? Good catch! 

To be consistent with method naming, I would prefer to remove line 35 and keep the new line 25.